android-project/app/src/main/java/oceantreasur/es/android_project/GameActivity.java

```diff
@@ -147,16 +126,14 @@ public void onClick(View v) {
 
```

Comment:
Don't point views towards a click listener through the xml. Define an anonymous OnClickListener object and pass it to all ImageViews in the first cycle (when you're initializing them). This way anyone reviewing the code will instantly know who's calling this onClick event.